gulpfile.babel.js

```diff
@@ -0,0 +1,153 @@
+import webpackConfig from './webpack.config.babel.js';
+import WebpackDevServer from 'webpack-dev-server';
+import eslint from 'gulp-eslint';
+import jscs from 'gulp-jscs';
+import header from 'gulp-header';
+import esdoc from 'gulp-esdoc';
+import path from 'path';
+
+const DEST = 'dist';
+const banner = [
+  '/**',
+  ` * @file ${pkg.name}`,
+  ` * @description ${pkg.description}`,
+  ` * @version v${pkg.version}`,
+  ` * @see {@link ${pkg.homepage}}`,
+  ' * @license MIT',
+  ` * @author ${pkg.author.name}`,
+  ` * @copyright ${new Date().getFullYear()} Krux Digital, Inc`,
+  ' */',
+  ''
+].join('\n');
+
+gulp.task('clean', () => {
+  return del.sync([`${DEST}/**`]);
+});
+
+const LINTABLE_PATTERN = ['*.js', 'src/**/*.js', 'test/unit/*.spec.js'];
+
+gulp.task('eslint', () => {
+  return gulp.src(LINTABLE_PATTERN)
+    .pipe(eslint())
+    .pipe(eslint.format())
+    .pipe(eslint.failAfterError());
+});
+
+gulp.task('jscs', () => {
+  return gulp.src(LINTABLE_PATTERN)
+    .pipe(jscs())
+    .pipe(jscs.reporter());
+});
+
+gulp.task('lint', ['eslint', 'jscs']);
+
+const basename = path.basename(webpackConfig.output.filename, path.extname(webpackConfig.output.filename));
+
+function build(config) {
+  return () => webpackStream(config)
+    .pipe(header(banner))
+    .pipe(rename({basename: basename, extname: '.max.js'}))
+    .pipe(gulp.dest(DEST))
+    .pipe(stripDebug())
+    .pipe(rename({basename: basename, extname: '.js'}))
+    .pipe(gulp.dest(DEST))
+    .pipe(uglify())
+    .pipe(header(banner))
+    .pipe(rename({basename: basename, extname: '.min.js'}))
+    .pipe(gulp.dest(DEST));
+}
+
+gulp.task('build', build(webpackConfig));
+
+gulp.task('watch', () => {
+  gulp.watch('src/**/*.js', ['build']);
+});
+
+gulp.task('serve', ['clean', 'build'], done => {
+  const devConfig = Object.create(webpackConfig);
+  devConfig.devtool = 'eval';
+  devConfig.debug = true;
+
+  gutil.log('[serve]', `http://localhost:${pkg.config.ports.cdn}/`);
+  new WebpackDevServer(webpack(devConfig), {
+    contentBase: DEST,
+    hot: true,
+    quiet: false,
+    stats: {
+      colors: true
+    }
+  }).listen(pkg.config.ports.cdn, 'localhost', (err, stats) => {
+    if (err) {
+      throw new gutil.PluginError('webpack', err);
+    }
+
+    if (stats) {
+      gutil.log('[serve]', stats.toString({
+        colors: true
+      }));
+    }
+
+    done();
+  });
+});
+
+function karma(configName, failOnError = true) {
+  return done => new Karma({
+    configFile: path.resolve(`./${configName}.config.babel.js`)
+  }, err => {
+    if (err) {
+      gutil.log('[test]', 'Tests failed');
+      if (failOnError) {
+        process.exit(err);
+      }
+    }
+    done();
+  }).start();
+}
+
+gulp.task('test', karma('karma'));
+
+gulp.task('test:cross-browser', karma('karma-sauce'));
+
+gulp.task('test:tdd', karma('karma', false));
+
+gulp.task('tdd', ['test:tdd'], () => {
+  gulp.watch(['*.js', 'src/**', 'test/**'], ['test:tdd']);
```

Comment:
I'd skip `*.js` here; it confusing to see the tests reload, but the config changes not take effect. (Assuming gulp doesn't reload the gulpfile in these watch statements.)